### PR TITLE
[MAINTENANCE] Tests to prove sqlite partitioners work

### PR DESCRIPTION
**Pull request by tyler-hoffman**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/integration/data_sources_and_expectations/data_sources/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+"""Tetss that focus on the capabilities and edge cases of specific data sources.
```
Comment (cdkini) [inline tests/integration/data_sources_and_expectations/data_sources/__init__.py]:
```suggestion
"""Tests that focus on the capabilities and edge cases of specific data sources.
```

File: tests/integration/data_sources_and_expectations/data_sources/test_sqlite.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    All tests use ExpectColumnDistinctValuesToEqualSet to detect that we are just seeing the
+    appropriate rows.
+    """
+
+    @parameterize_batch_for_data_sources(
+        data_source_configs=JUST_SQLITE,
+        data=pd.DataFrame(DATA),
+    )
+    def test_yearly_partitioning(self, asset_for_datasource: TableAsset) -> None:
+        batch_def = asset_for_datasource.add_batch_definition_yearly("yearly", column="date")
```
Comment (cdkini) [inline tests/integration/data_sources_and_expectations/data_sources/test_sqlite.py]: Nit - if using constants elsewhere, could we make `column=VALUE_COL`?

**Review by cdkini — APPROVED** [review]

File: tests/integration/data_sources_and_expectations/data_sources/__init__.py (hunk 1 above)
Comment (tyler-hoffman) [reply tests/integration/data_sources_and_expectations/data_sources/__init__.py]: ugh typos in docstrings are my signature move

File: tests/integration/data_sources_and_expectations/data_sources/test_sqlite.py (hunk 2 above)
Comment (tyler-hoffman) [reply tests/integration/data_sources_and_expectations/data_sources/test_sqlite.py]: Good call. Updated the next couple tests too
